### PR TITLE
Add Azure connection handling

### PR DESCRIPTION
**Pull request by rick-a-lane-ii**
Add Azure connection handling

### CODE REVIEW

File: pkg/task/azure.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+	if directory == "" {
+		directory = filepath.Join(DefaultPath, ".azure")
+	}
+
+	// .azure/credentials
+	creds := ini.Empty()
+	creds.Section(spec.Azure.Connection.SubscriptionID).Key("client_id").SetValue(spec.Azure.Connection.ClientID)
+	creds.Section(spec.Azure.Connection.SubscriptionID).Key("secret").SetValue(spec.Azure.Connection.Secret)
+	creds.Section(spec.Azure.Connection.SubscriptionID).Key("tenant").SetValue(spec.Azure.Connection.TenantID)
```
Comment (hunner) [inline pkg/task/azure.go:32]: I think we want all four of `subscription_id`, `client_id`, `secret`, and `tenant` under a "profile" section. The azure cli will look for a `default` profile by default, but other profiles can have arbitrary names.

Maybe you have docs that describe using the subscription ID as the section name? I couldn't find a mention of that yet.

**Review by hunner — COMMENTED** [review]
.

File: pkg/task/azure.go (hunk 1 above)
Comment (hunner) [reply pkg/task/azure.go:32]: Oh, I do see https://dev.to/azure/setting-up-the-azure-credentials-file-41o3 now... looking for a more definitive source

Comment (hunner) [reply pkg/task/azure.go:32]: Not even right; the format with `[default]` is for `ansible` not `az`. Whoops.

Comment (rick-a-lane-ii) [reply pkg/task/azure.go:32]: Yeah, that's what I based it off of. Testing it out.

**Review by hunner — APPROVED** [review]
🔷

I can't find any official documentation on the format of the credentials file in `AZURE_CONFIG_DIR` but assume that what you have is the right format. 👍🏻

File: pkg/task/azure.go (hunk 1 above)
Comment (rick-a-lane-ii) [reply pkg/task/azure.go]: That information was either outdated or incorrect...

**Pull request merged**